StabilityPackage.wl

```diff
@@ -211,18 +357,112 @@ Reap[
++ 3 z datasr["lotsodo"][[1,E,2]]+(5/2 (3 (datasr["lotsodo"][[1,E,1]] -datasr["lotsodo"][[1,E,3]] )/2 x^2+3 (datasr["lotsodo"][[1,E,1]] -datasr["lotsodo"][[1,E,3]] )/2 y^2+3 datasr["lotsodo"][[1,E,3]] z^2-datasr["lotsodo"][[1,E,1]])));
+
+afoc1234[x_,y_,z_,E_]:= ((3c^3)/(4 Pi h (1/2 (data["freq"][[E+1]]+data["freq"][[E]])) (data["freq"][[E+1]]^3-data["freq"][[E]]^3)) )(datasr["lotsodo"][[2,E,1]] + 3 z datasr["lotsodo"][[2,E,2]]
++(5/2 (3 (datasr["lotsodo"][[2,E,1]] -datasr["lotsodo"][[2,E,3]] )/2 x^2+3 (datasr["lotsodo"][[2,E,1]] -datasr["lotsodo"][[2,E,3]] )/2 y^2+3 datasr["lotsodo"][[2,E,3]] z^2-datasr["lotsodo"][[2,E,1]] )));
+
+xfoc1234[x_,y_,z_,E_]:= ((3c^3)/(4 Pi h (1/2 (data["freq"][[E+1]]+data["freq"][[E]])) (data["freq"][[E+1]]^3-data["freq"][[E]]^3)) )(0.25 datasr["lotsodo"][[3,E,1]] + 3 z 0.25 datasr["lotsodo"][[3,E,2]]
++(5/2 (3 (0.25 datasr["lotsodo"][[3,E,1]] - 0.25 datasr["lotsodo"][[3,E,3]] )/2 x^2+3 (0.25 datasr["lotsodo"][[3,E,1]] - 0.25 datasr["lotsodo"][[3,E,3]] )/2 y^2+3 0.25 datasr["lotsodo"][[3,E,3]] z^2- 0.25 datasr["lotsodo"][[3,E,1]] )));
+
+
+ei[m_]:= Sum[1/3 (datasr["freq"][[f+1]]^3-datasr["freq"][[f]]^3)foc1234[Sin[ArcCos[m]],0,m,f],{f,1,80}];
+aei[m_]:= Sum[1/3 (datasr["freq"][[f+1]]^3-datasr["freq"][[f]]^3)afoc1234[Sin[ArcCos[m]],0,m,f],{f,1,80}];
+xei[m_]:= Sum[1/3 (datasr["freq"][[f+1]]^3-datasr["freq"][[f]]^3)xfoc1234[Sin[ArcCos[m]],0,m,f],{f,1,80}];
+ag=1/2 (Abs[ei[1]]+Abs[ei[-1]]);
+aag=1/2 (Abs[aei[1]]+Abs[aei[-1]]);
+xag = 1/2 (Abs[xei[1]]+Abs[xei[-1]]);
+\[Beta]g=0;
+\[Chi]g=-5;
+a\[Beta]g=0;
+a\[Chi]g=-5;
+(*returns the appropriate guesses based on the species*)
+Which[
+species==1, g0={ag,\[Beta]g,\[Chi]g},
+
+species==2, g0={aag,a\[Beta]g,a\[Chi]g},
+species==3, g0={xag,\[Beta]g,\[Chi]g}
+
+];
+Return[g0];
+]
+
+eBoxFitSingleRadius[data_,ri_,species_,guesses_]:=Module[{ebox,es1box,es2box,es3box,ne,Fe,Pre,ane,aFe,aPre,datasr,br,g0},
+datasr=SelectSingleRadius[data,ri];
+(*If this is the first radial bin, generates intial guesses based on the expansion reconstruction of the distribution function for a. \[Beta] and \[Chi] are set to intial guesses based on previous experience.*)
+If[
+ri== 1,
+g0=getInitialGuess[data,species],
+,g0=guesses;
+];
+
+(*Equation of an ellipse in the "box transform" units*)
+ebox[a_,\[Beta]_,\[Chi]_,m_]:=(a (1+Tanh[\[Beta]]) (1/4 a^2 m (1+Tanh[\[Beta]]) (1+Tanh[\[Chi]])
++a Sqrt[-a^2 (-1+m^2)+1/4 a^2 m^2 (1+Tanh[\[Beta]])^2+1/4 a^2 (-1+m^2) (1+Tanh[\[Chi]])^2]))/(2 (a^2+m^2 (-a^2+1/4 a^2 (1+Tanh[\[Beta]])^2)));
+es1box[a_,\[Beta]_,\[Chi]_]:=1/c^3 NIntegrate[ebox[a,\[Beta],\[Chi],m],{m,-1.,1.},MaxRecursion->13];
+es2box[a_,\[Beta]_,\[Chi]_]:=1/c^3 NIntegrate[m ebox[a,\[Beta],\[Chi],m],{m,-1.,1.},MaxRecursion->13];
+es3box[a_,\[Beta]_,\[Chi]_]:=1/c^3 NIntegrate[ m^2 ebox[a,\[Beta],\[Chi],m],{m,-1.,1.},MaxRecursion->13];
+
+Which[
+species==1,
+ne=Sum[ datasr["lotsodo"][[1,f,1]]/( h (1/2 (data["freq"][[f+1]]+data["freq"][[f]]))),{f,1,80}];
```

Comment:
Again here, would be better to define a function that is passed a species index to prevent repeated code

Reply:
Ah, this function already is passed species. So here, you can put the weight inside of the which, and have the complicated expressions outside of the which and just use "weight" as a prefactor and "species" as the index

Reply:
Oh I see, got it.